src/js/game/hud/parts/building_placer_logic.js

```diff
@@ -44,7 +44,32 @@ export class HUDBuildingPlacerLogic extends BaseHUDPart {
          * The current rotation
          * @type {number}
          */
-        this.currentBaseRotation = 0;
+        this.currentBaseRotationGeneral = 0;
+
+        /**
+         * The current rotation preference for each building.
+         * @type {Object.<string, number>}
+         */
+        this.preferredRotations = {};
+
+        this.getBaseRotation = function () {
```

Comment:
The game uses es6 classes, could you convert that into es6 class functions? Or am I missing something here?